Task_4/scripts/app.js

```diff
@@ -0,0 +1,377 @@
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  function getTasks(skip = 0, top = 10, filterConfig) {
+    try {
+      if (
+        typeof skip !== "number" ||
+        typeof top !== "number" ||
+        !isFinite(skip) ||
+        !isFinite(top)
+      ) {
+        throw new Error(ERRORS.invalidValue);
+      }
+
+      let result = [...tasks].sort(
+        (a, b) => Date.parse(a.createdAt) - Date.parse(b.createdAt)
+      );
+
+      for (key in filterConfig) {
+        result = result.filter((elem) => {
+          if (key === KEYS.name || key === KEYS.description) {
+            return elem[key]
+              .toLowerCase()
+              .includes(filterConfig[key].toLowerCase());
+          }
+          if (
+            key === KEYS.assignee ||
+            key === KEYS.status ||
+            key === KEYS.priority
+          ) {
+            return elem[key].toLowerCase() === filterConfig[key].toLowerCase();
+          }
+          if (key === KEYS.isPrivate) {
+            return elem[key] === filterConfig[key];
+          }
+          if (key === KEYS.dateFrom) {
+            return Date.parse(elem.createdAt) >= Date.parse(filterConfig[key]);
+          }
+          if (key === KEYS.dateTo) {
+            return Date.parse(elem.createdAt) <= Date.parse(filterConfig[key]);
+          }
+        });
+      }
+
+      return result.splice(skip, top);
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  function changeUser(usr) {
+    try {
+      if (usr === user.name) {
+        throw new Error(ERRORS.sameName);
+      }
+      if (typeof usr !== "string" || usr.trim().length === 0) {
+        throw new Error(ERRORS.invalidValue);
+      }
+
+      user.setName = usr;
+
+      return true;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  function validateComment(com) {
+    try {
+      if (typeof com !== "object" || com === null || Array.isArray(com)) {
+        throw new Error(ERRORS.invalidValue);
+      }
+
+      if (!Object.keys(com).length) {
+        return true;
+      }
+
+      if (Object.keys(com).length !== MAX_LENGTH.com) {
+        throw new Error(ERRORS.lengthNotValidate);
+      }
+
+      if (
+        !validateObj.id(com.id) ||
+        !validateObj.description(com.text) ||
+        !validateObj.createdAt(com.createdAt) ||
+        !validateObj.assignee(com.author)
+      ) {
+        throw new Error(ERRORS.valuesNotValidate);
+      }
+
+      return true;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  }
+
+  function addComment(idTask, text) {
+    try {
+      if (!isValidTypeId(idTask) || !findTaskById(idTask)) {
+        throw new Error(ERRORS.taskNotFound);
+      }
+
+      if (!validateObj.description(text)) {
+        throw new Error(ERRORS.descriptionNotValidate);
+      }
+
+      const com = {
+        id: String(new Date().valueOf()),
```

Comment:
better use getTime(), because valueOf method is usually called internally by JavaScript and not explicitly in code.